src/main/java/org/broadinstitute/hellbender/engine/AutoDiscoverable.java

```diff
@@ -0,0 +1,10 @@
+package org.broadinstitute.hellbender.engine;
+
+/**
+ * This marker interface indicates that the instanceof the marked class are 'magically'
+ * loaded at runtime via the {@link org.broadinstitute.hellbender.cmdline.ClassFinder}.
+ * New instances of each such class will be created via reflection
+ * and therefore each class that implements this interface must have a public no-args constructor.
+ */
+public interface AutoDiscoverable {
```

Comment:
`FeatureCodecs` are also auto-discovered by us (in `FeatureManager`), but we can't mark it with the new `AutoDiscoverable` interface since it lives in htsjdk. What do you suggest we do about this? (at least we should mention this in the docs for `AutoDiscoverable`)